mapper/thin_edge_json/src/json.rs

```diff
@@ -282,12 +276,102 @@ impl ThinEdgeJsonError {
 
 mod tests {
     use super::*;
+    use crate::measurement::*;
+    use assert_matches::*;
 
     fn test_timestamp() -> DateTime<FixedOffset> {
         FixedOffset::east(5 * 3600)
             .ymd(2021, 04, 08)
             .and_hms(0, 0, 0)
     }
+    #[test]
+    fn parse_from_rfc3339_with_empty_timestamp() {
+        assert_matches!(
+            ThinEdgeJson::parse_from_rfc3339("...."),
+            Err(ThinEdgeJsonError::InvalidTimestamp { .. })
+        );
+    }
+    #[test]
+    fn parse_from_rfc3339_timestamp_with_seconds() {
+        assert_eq!(
+            ThinEdgeJson::parse_from_rfc3339("2021-04-08T00:00:00+05:00").unwrap(),
+            FixedOffset::east(5 * 3600)
+                .ymd(2021, 04, 08)
+                .and_hms(0, 0, 0)
+        );
+    }
+    #[test]
+    fn parse_from_rfc3339_timestamp_with_milliseconds() {

```

Comment:
drop this test case. no need to test `DateTime::parse_from_rfc3339`

Reply:
removed